### PR TITLE
[EigenSolvers] Add Eigen library to application

### PR DESCRIPTION
**Pull request by oberbichler**
As discussed yesterday the Eigen library is now included inside the `EigenSolversApplication`.

Tested on windows.

Can you do a quick test if this is also working on your machines?

### CODE REVIEW

**Comment by qaumann** [comment]
Cool, works for me on Linux

File: applications/EigenSolversApplication/CMakeLists.txt [hunk 1]
```diff
@@ -16,10 +16,6 @@ message( "**** configuring KratosEigenSolversApplication ****" )
 ################### PYBIND11
 include(pybind11Tools)
 
-if( NOT DEFINED EIGEN_ROOT )
```
Comment (armingeiser) [inline applications/EigenSolversApplication/CMakeLists.txt:19]: As the EIGEN_ROOT variable is not needed anymore, the README of the application has to be adapted.

**Review by armingeiser — CHANGES_REQUESTED** [review]

File: applications/EigenSolversApplication/CMakeLists.txt (hunk 1 above)
Comment (armingeiser) [reply applications/EigenSolversApplication/CMakeLists.txt:19]: done

**Review by armingeiser — APPROVED** [review]

**Review by philbucher — CHANGES_REQUESTED** [review]
blocking to prevent accidental merge

**Comment by philbucher** [comment]
@KratosMultiphysics/technical-committee approves this, thanks for the effort

we will leave the treatment of `DEIGEN_MPL2_ONLY` as is for the moment, unless someone needs it, in which case please let us know

**Review by philbucher — APPROVED** [review]

**Comment by philbucher** [comment]
for the record, this added 5,1MB in files

**Pull request merged**